### PR TITLE
[test] Test more alignment for load extend and splat ops

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@AndrewScheidecker, @tlively, PTAL, thanks!

File: Test/spec/simd/simd_align.wast [hunk 1]
```diff
@@ -1,116 +1,307 @@
-;; Vaild alignment (align=1, 2, 4, 8, 16)
```
Comment (AndrewScheidecker) [inline Test/spec/simd/simd_align.wast]: s/Vaild/Valid

File: Test/spec/simd/simd_align.wast [hunk 2]
```diff
@@ -1,116 +1,307 @@
...
+    "(memory 0) (func (v128.store align=7 (i32.const 0) (v128.const i32x4 0 0 0 0)))"
   )
-  (func (export "v128.store_align_2") (result v128)
-    (v128.store align=2 (i32.const 0) (v128.const i32x4 0 1 2 3))
-    (v128.load (i32.const 0))
+  "alignment must be a power of two"
+)
+(assert_malformed
+  (module quote
+    "(memroy 1) (func (result v128) (i16x8.load8x8_s align=-1 (i32.const 0)))"
```
Comment (AndrewScheidecker) [inline Test/spec/simd/simd_align.wast]: s/memroy/memory here and elsewhere in the file

File: Test/spec/simd/simd_align.wast [hunk 3]
```diff
@@ -161,4 +352,4 @@
 (assert_return (invoke "v128_unalign_read_and_write") (v128.const i8x16 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15))
 (assert_return (invoke "v128_aligned_read_and_write") (v128.const i16x8 0 1 2 3 4 5 6 7))
 (assert_return (invoke "v128_aligned_read_and_unalign_write") (v128.const i32x4 0 1 2 3))
-(assert_return (invoke "v128_unalign_read_and_aligned_write") (v128.const i32x4 0 1 2 3))
+(assert_return (invoke "v128_unalign_read_and_aligned_write") (v128.const i32x4 0 1 2 3))
```
Comment (tlively) [inline Test/spec/simd/simd_align.wast]: Not from this PR, but "unalign" in this name should be "unaligned" as well.

**Comment by Honry** [comment]
@tlively, @AndrewScheidecker, thanks for catching these stupid typos, all fixed.

**Comment by AndrewScheidecker** [comment]
LGTM, thanks!

**Pull request merged**